rosbag2_cpp/src/rosbag2_cpp/rmw_implemented_serialization_format_converter.cpp

```diff
@@ -0,0 +1,132 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "./rmw_implemented_serialization_format_converter.hpp"
+
+#include <memory>
+#include <sstream>
+#include <string>
+
+#include "ament_index_cpp/get_resources.hpp"
+#include "rcpputils/find_library.hpp"
+#include "rcpputils/shared_library.hpp"
+#include "rmw/rmw.h"
+#include "rosbag2_cpp/converter_interfaces/serialization_format_converter.hpp"
+#include "rosbag2_cpp/logging.hpp"
+
+namespace
+{
+/// Get a function with a given signature from a shared library, with error checking.
+template<typename T>
+T get_function_from(const char * function_name, std::shared_ptr<rcpputils::SharedLibrary> library)
+{
+  if (!library->has_symbol(function_name)) {
+    std::stringstream ss;
+    ss << "Converter could not find expected symbol '" << function_name <<
+      "' in rmw implementation library " << library->get_library_path();
+    throw std::runtime_error{ss.str()};
+  }
+  T loaded_function = nullptr;
+  // Function expected to return a value because has_symbol was checked first.
+  loaded_function = (decltype(loaded_function))(library->get_symbol(function_name));
+  return loaded_function;
+}
+}  // namespace
+
+namespace rosbag2_cpp
+{
+
+class RMWImplementedConverterImpl
```

Comment:
I can't help it, but that name somewhat sounds complicated to me. Unfortunately, I also don't have a proposal for a better name :P